crates/ra_ide_api_light/src/assists/introduce_match.rs

```diff
@@ -0,0 +1,116 @@
+use ra_syntax::TextUnit;
+
+use crate::assists::{AssistCtx, Assist};
+
+pub fn introduce_match<'a>(ctx: AssistCtx) -> Option<Assist> {
+    let expr = ctx.covering_node();
```

Comment:
This will make the assist always available. 

At minimum, I think we need to check that we have an expression at the cursor. 

```
let expr = node.ancestors().filter_map(ast::Expr::cast).next()?;
```